utils/keys_test.go

```diff
@@ -315,21 +314,50 @@ func TestNonRootPathInference(t *testing.T) {
 	}
 }
 
-func TestBlockHeaderPrecedence(t *testing.T) {
+func TestBlockHeaderPrecedenceRoleAndGun(t *testing.T) {
 	s := NewTestImportStore()
 
-	b := &pem.Block{
-		Headers: make(map[string]string),
+	from, _ := os.OpenFile("../fixtures/secure.example.com.key", os.O_RDONLY, notary.PrivKeyPerms)
+	defer from.Close()
+	fromBytes, _ := ioutil.ReadAll(from)
+	b, _ := pem.Decode(fromBytes)
+	b.Headers["role"] = data.CanonicalSnapshotRole
+	b.Headers["gun"] = "anothergun"
+	bBytes := pem.EncodeToMemory(b)
+
+	in := bytes.NewBuffer(bBytes)
+
+	err := ImportKeys(in, []Importer{s}, "somerole", "somegun", passphraseRetriever)
+	require.NoError(t, err)
+
+	require.Len(t, s.data, 1)
+	for key := range s.data {
+		// block header role= root should take precedence over command line flag
+		require.Equal(t, key, filepath.Join(notary.NonRootKeysSubdir, "anothergun", "12ba0e0a8e05e177bc2c3489bdb6d28836879469f078e68a4812fc8a2d521497"))
+		final, rest := pem.Decode(s.data[key])
+		require.Len(t, rest, 0)
+		require.Equal(t, final.Headers["role"], "snapshot")
+		require.Equal(t, final.Headers["gun"], "anothergun")
 	}
-	from, _ := os.OpenFile("../fixtures/precedence.example.com.key", os.O_RDONLY, notary.PrivKeyPerms)
+}
+
+func TestBlockHeaderPrecedencePathGun(t *testing.T) {
```

Comment:
Non-blocking nitpick:  should this be "TestBlockHeaderPrecedencePathRole", since the header has both the path and the role, but not the gun?